src/mcts/search.cc

```diff
@@ -1237,15 +1237,15 @@ void SearchWorker::ExtendNode(Node* node) {
           m = std::max(0.0f, parent->GetM() - 1.0f);
         }
         // If the colors seem backwards, check the checkmate check above.
-        if (wdl == WDL_WIN) {
-          node->MakeTerminal(GameResult::BLACK_WON, m,
-                             Node::Terminal::Tablebase);
-        } else if (wdl == WDL_LOSS) {
-          node->MakeTerminal(GameResult::WHITE_WON, m,
-                             Node::Terminal::Tablebase);
-        } else {  // Cursed wins and blessed losses count as draws.
-          node->MakeTerminal(GameResult::DRAW, m, Node::Terminal::Tablebase);
-        }
+        const auto use_50_move_rule = params_.GetSyzygy50MoveRule();
+        // Cursed wins and blessed losses also are draws using 50 move rule.
+        node->MakeTerminal(
+            wdl >= (use_50_move_rule ? WDL_WIN : WDL_CURSED_WIN)
+                ? GameResult::BLACK_WON
+                : wdl <= (use_50_move_rule ? WDL_LOSS : WDL_BLESSED_LOSS)
+                      ? GameResult::WHITE_WON
+                      : GameResult::DRAW,
```

Comment:
The optimizer ends up using `!condition ^ 3` to get 3 or 2 argument value… 😝 
https://github.com/LeelaChessZero/lc0/blob/c405b15d3f383c2d077f73ff5d3a2052142e0b9e/src/chess/position.h#L81